src/engraving/libmscore/repeatlist.cpp

```diff
@@ -1027,6 +1031,49 @@ void RepeatList::unwind()
     }
 
     updateTempo();
-    _expanded = true;
+    m_expanded = true;
+}
+
+const Measure* RepeatList::playbackMeasure(size_t staffIdx, const Measure* measure) const
+{
+    if (m_measureRepeatsMapping.find(staffIdx) == m_measureRepeatsMapping.cend()) {
+        return measure;
+    }
+
+    const MeasureRepeatsStaffMapping& staffMapping = m_measureRepeatsMapping.at(staffIdx);
+    if (staffMapping.find(measure) == staffMapping.cend()) {
+        return measure;
+    }
+
+    return staffMapping.at(measure);
+}
+
+void RepeatList::updateMeasureRepeatsMapping()
+{
+    m_measureRepeatsMapping.clear();
+
+    size_t nstaves = m_score->nstaves();
+    for (size_t staffIdx = 0; staffIdx < nstaves; ++staffIdx) {
+        MeasureRepeatsStaffMapping& staffMapping = m_measureRepeatsMapping[staffIdx];
+
+        for (const Measure* measure = m_score->firstMeasure(); measure; measure = measure->nextMeasure()) {
+            if (measure->isMeasureRepeatGroup(staffIdx)) {
+                const Measure* playbackMeasure = measure;
+                for (int i = 0; playbackMeasure && i < measure->measureRepeatNumMeasures(staffIdx); i++) {
+                    playbackMeasure = playbackMeasure->prevMeasure();
+                }
+
+                if (playbackMeasure) {
+                    if (staffMapping.find(playbackMeasure) != staffMapping.cend()) {
+                        playbackMeasure = staffMapping.at(playbackMeasure);
+
+                        Q_ASSERT(staffMapping.find(playbackMeasure) == staffMapping.cend());
```

Comment:
always prefer to use the one from log.h

Reply:
There is not just an assert function in log.h; only IF_ASSERT_FAILED etc., but that is pointless in this case.
Also, Q_ASSERT is used in numerous places, also in new code, and also as the 'backend' of the things in log.h. 
Of course, we can replace all Q_ASSERT with something like MU_ASSERT which we #define to Q_ASSERT, but that's a separate job. Shall I do that?